pgml-extension/src/orm/snapshot.rs

```diff
@@ -811,19 +829,217 @@ impl Snapshot {
                     }
                 }
             });
+            let num_distinct_labels = class_train.iter().cloned().collect::<HashSet<_>>().len();
+            data = Some(TextClassificationDataset {
+                text_train,
+                class_train,
+                text_test,
+                class_test,
+                num_features,
+                num_labels,
+                num_rows,
+                num_test_rows,
+                num_train_rows,
+                // TODO rename and audit this
+                num_distinct_labels,
+            });
 
-            data = Some(TextDataset {
-                x_train,
-                y_train,
-                x_test,
-                y_test,
+            Ok::<std::option::Option<()>, i64>(Some(())) // this return type is nonsense
+        })
+        .unwrap();
+
+        let data = data.unwrap();
+
+        info!("{}", data);
+
+        data
+    }
+
+    pub fn text_pair_classification_dataset(
+        &mut self,
+        dataset_args: default!(JsonB, "'{}'"),
+    ) -> TextPairClassificationDataset {
+        let mut data = None;
+
+        Spi::connect(|client| {
+            let result = client.select(&self.select_sql(), None, None).unwrap();
+            let num_rows = result.len();
+            let (num_train_rows, num_test_rows) = self.train_test_split(num_rows);
+            let num_features = 2;
+            let num_labels = self.num_labels();
+
+            let mut text1_train: Vec<String> = Vec::with_capacity(num_train_rows);
+            let mut text2_train: Vec<String> = Vec::with_capacity(num_train_rows);
+            let mut class_train: Vec<String> = Vec::with_capacity(num_train_rows);
+            let mut text1_test: Vec<String> = Vec::with_capacity(num_test_rows);
+            let mut text2_test: Vec<String> = Vec::with_capacity(num_test_rows);
+            let mut class_test: Vec<String> = Vec::with_capacity(num_test_rows);
+
+            let text1_column_value = dataset_args
+                .0
+                .get("text1_column")
```

Comment:
do we require these column names?

Reply:
Yes, for text pair classification - (natural language inference, qnli etc.), we need three columns - text1, text2 and the class.